dask/dataframe/core.py

```diff
@@ -4328,18 +4328,27 @@ def set_index(
 
         Examples
         --------
-        >>> df2 = df.set_index('x')  # doctest: +SKIP
-        >>> df2 = df.set_index(d.x)  # doctest: +SKIP
-        >>> df2 = df.set_index(d.timestamp, sorted=True)  # doctest: +SKIP
+        >>> ddf = dask.datasets.timeseries(start="2021-01-01", end="2021-01-07", freq="1H").reset_index()
+        >>> ddf2 = ddf.set_index("x")
+        >>> ddf2 = ddf.set_index(ddf.x)
+        >>> ddf2 = ddf.set_index(ddf.timestamp, sorted=True)
 
         A common case is when we have a datetime column that we know to be
         sorted and is cleanly divided by day.  We can set this index for free
         by specifying both that the column is pre-sorted and the particular
         divisions along which is is separated
 
         >>> import pandas as pd
-        >>> divisions = pd.date_range('2000', '2010', freq='1D')
-        >>> df2 = df.set_index('timestamp', sorted=True, divisions=divisions)  # doctest: +SKIP
+        >>> divisions = pd.date_range(start="2021-01-01", end="2021-01-07", freq='1D')
+        ... divisions
+        DatetimeIndex(['2021-01-01', '2021-01-02', '2021-01-03', '2021-01-04',
+                    '2021-01-05', '2021-01-06', '2021-01-07'],
+                    dtype='datetime64[ns]', freq='D')
+
+        Note that ``len(divisons)`` is equal to ``npartitions + 1``. This is because ``divisions``
+        represents the upper and lower bounds of each partition.
+
+        >>> ddf2 = ddf.set_index("timestamp", sorted=True, divisions=divisions)
```

Comment:
I wonder if the example would be easier to read if we used the `name` column and wrote the divisions by hand:
```python
        >>> divisions = ["Alice", "Frank", "Laura", "Quinn", "Ursula", "Zelda"]
        >>> ddf2 = ddf.set_index("name", divisions=divisions)
```
This doesn't illustrate `sorted=True`, but it does show a probably-more-common case of using a different unsorted column, and that writing divisions yourself is not that scary.

Reply:
Yeah I like that idea.